### PR TITLE
A sqlite-backend QuoteBook implementation

### PR DESCRIPTION
**Pull request by eranrund**
This allows us to have the quote book persisted and survive restarts of the server.
This includes a bugfix (and an updated unit test) for the case of input rules with a 0 max tombstone block.

### CODE REVIEW

**Comment by eranrund** [comment]
Current dependencies on/for this PR:
* main
  * **PR #20** <a href="https://app.graphite.dev/github/pr/mobilecoinofficial/deqs/20" target="_blank">Graphite</a>  👈
    * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinofficial/deqs/21" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinofficial/deqs/20?utm_source=stack-comment).

**Review by cbeck88 — APPROVED** [review]

File: quote-book/sqlite/migrations/2023-02-17-204631_quotes/up.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    -- The number of counter tokens needed to trade the max amount of base tokens.
+    max_counter_tokens BINARY NOT NULL,
+
+    -- Timestamp at which the quote arrived
+    timestamp BIGINT NOT NULL,
+
+    -- The key image of the quote.
+    key_image BINARY NOT NULL,
+
+    -- The tombstone block of the quote
```
Comment (wjuan-mob) [inline quote-book/sqlite/migrations/2023-02-17-204631_quotes/up.sql:40]: Don't we do inequality comparisons with the tombstone block? Should this also be in Binary for the comparisons to work for unsigned?

Comment (cbeck88) [reply quote-book/sqlite/migrations/2023-02-17-204631_quotes/up.sql:40]: i think if we actually have more than `i64::Max` blocks in the mobilecoin blockchain, we are all going to be either sipping margaritas on a beach somewhere, or else, dead 500 years ago

Comment (cbeck88) [reply quote-book/sqlite/migrations/2023-02-17-204631_quotes/up.sql:40]: but like, you're not wrong, we could make it binary i guess

Comment (eranrund) [reply quote-book/sqlite/migrations/2023-02-17-204631_quotes/up.sql:40]: yeah i dont think we need to worry about exceeding i64 block count

**Pull request merged**